src/Encryption/Nip44.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+declare(strict_types=1);
+
+namespace swentel\nostr\Encryption;
+
+use Elliptic\EC;
+use Exception;
+use ParagonIE\Sodium\Compat;
+
+/**
+ * NIP-44 encryption implementation.
+ * Based on the reference implementation from nostr-tools.
+ */
+class Nip44
+{
+    private const VERSION = 2;
+    private const MIN_PLAINTEXT_SIZE = 1;
+    private const MAX_PLAINTEXT_SIZE = 0xffff;
+
+    /**
+     * Get conversation key using HKDF with shared secret.
+     */
+    public static function getConversationKey(string $privateKey, string $publicKey): string
+    {
```

Comment:
If $privateKey and/or $publicKey are bech32 formatted:

1. convert to hex formatted key
2. throw Exception